can_tools/scrapers/official/base.py

```diff
@@ -121,6 +121,12 @@ def _put_exec(self, engine: Engine, df: pd.DataFrame) -> None:
                 rows_deleted = res_delete.rowcount
                 print("Removed the {} rows from temp table".format(rows_deleted))
 
+        if rows_inserted == 0:
```

Comment:
Are we confident that we never intentionally write 0 rows?  If we were worried about that we could add a check for `df.shape[0] == 0` here or at the top of the method or something.

Reply:
I'm not 100% confident, but it seems that we should never need to. I've extended this `if` in f66c34f53240642436abab23cb2542357b461f0c to not throw the error if `df` is actually empty